### DISCUSSION TITLE
Invariance of solutions

### DISCUSSION THREAD

**Discussion by clizbe** [TulipaEnergy/TulipaEnergyModel.jl#623] [Ideas]
(Something akin to property based testing)
@suvayu Do you want to kick-off the discussion?

**Comment by suvayu** [comment]
The idea is to test "invariant concepts", which may be a property of the solution that holds true for all (or a subset) of cases.  Typically this is probably something derived from a conservation law, or disallowing unphysical solutions, etc.  In principle there are probably many such properties, but not all of them are useful to check.  So if we check for properties that can be indication of various common errors, that would be very useful to provide feedback to the user.

A couple of (not-so-concrete) examples:
- a common modelling error that leads to unphysical solutions
- a data error, say mismatching units, or unintended mismatched resolution
